### PR TITLE
Seal private nested classes to avoid unneeded virtual methods

### LINKED ISSUE #19066: Seal private classes
Unless they are being inherited from in the same outer class. This will avoid having virtual methods on classes that cannot be inherited from.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #19066

Also removed `GetTypeInfo` when it is not needed

### CODE REVIEW

File: src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.EntityFrameworkCore.Diagnostics
     public static class AbstractionsStrings
     {
         private static readonly ResourceManager _resourceManager
-            = new ResourceManager("Microsoft.EntityFrameworkCore.Properties.AbstractionsStrings", typeof(AbstractionsStrings).GetTypeInfo().Assembly);
+            = new ResourceManager("Microsoft.EntityFrameworkCore.Properties.AbstractionsStrings", typeof(AbstractionsStrings).Assembly);
```
Comment (AndriySvyryd) [inline src/EFCore.Abstractions/Properties/AbstractionsStrings.Designer.cs]: You need to change the template as well

File: src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs [hunk 2]
```diff
@@ -168,7 +168,7 @@ private void SetValue(int index, object value)
 
             if (value != null)
             {
-                if (!property.ClrType.GetTypeInfo().IsAssignableFrom(value.GetType().GetTypeInfo()))
+                if (!property.ClrType.IsAssignableFrom(value.GetType()))
```
Comment (AndriySvyryd) [inline src/EFCore/ChangeTracking/Internal/ArrayPropertyValues.cs]: I remember that doing this on mono/xamarin without `GetTypeInfo()` didn't work. Hopefully that's fixed now.

File: src/EFCore/Query/QueryTranslationPreprocessor.cs [hunk 3]
```diff
@@ -41,7 +41,7 @@ public virtual Expression Process(Expression query)
         }
 
         // TODO: For debugging
-        private class EnumerableVerifyingExpressionVisitor : ExpressionVisitor
+        private sealed class EnumerableVerifyingExpressionVisitor : ExpressionVisitor
```
Comment (AndriySvyryd) [inline src/EFCore/Query/QueryTranslationPreprocessor.cs]: @smitpatel Do you remember what this comment means and whether it is still relevant?

Comment (smitpatel) [reply src/EFCore/Query/QueryTranslationPreprocessor.cs]: We can remove this class. It was to make sure we don't have enumerable methods where queryable should be there. But I believe exception from here won't be useful either way.

**Pull request merged**